### DISCUSSION TITLE
Using an option picker label instead of the value

### DISCUSSION THREAD

**Discussion by FlaminWrap** [Budibase/budibase#10950] [Help] [answered]
Hello,

I have a custom source option picker with various values in it to change variables in data sources to easily check different areas of activities that have taken place. An example of this is:

| Label  | Value |
| ------------- | ------------- |
| All Areas | %  |
| Area 1  | ABC  |
| Area 2  | DEF  |

I would like to show the label of the option picker selection in a paragraph but I only get the value. e.g. When All Areas id select it shows `All Activities in %` instead of `All Activities in All Areas`. Is there a way to show the label?

**Comment by ConorWebb96** [comment, marked as answer]
Hey @FlaminWrap,

Currently you are only able to set the value as I'm sure you have been doing. I'd recommend creating a enhancement request if you want to expand this functionality to include the labels as well.

With there being so few labels might I suggest something like this for a potential solution?

1. Update state on the options picker change.
<img width="100%" alt="Screenshot 2023-06-19 at 13 38 22" src="https://github.com/Budibase/budibase/assets/126772285/18b0e63f-81da-4201-9ac3-a0fa3591a307">
2. Use the below function to output the labels.
<img width="100%" alt="Screenshot 2023-06-19 at 13 38 36" src="https://github.com/Budibase/budibase/assets/126772285/5a07d7fd-82aa-4b72-8c52-76c47c4e7a80">

```
var fields = [
  { value: "%", label: "All Areas" },
  { value: "ABC", label: "Area 1" },
  { value: "DEF", label: "Area 2" }
];
var selectedValue = $("State.output");
var matchingField = fields.find(function(field) {
  return field.value === selectedValue;
});
if (matchingField) {
  return matchingField.label;
}
```

**Result**
![Loom Message - 19 June 2023](https://github.com/Budibase/budibase/assets/126772285/9226d646-f92f-4b8f-b946-a3f112f5f014)

**Comment by FlaminWrap** [comment]
Excellent, thank you.
